core/app/models/spree/order_contents.rb

```diff
@@ -83,6 +84,20 @@ def reload_totals
       @order.recalculate
     end
 
+    def destroy_non_positive_line_items(params)
+      line_items_to_destroy_ids = []
+      # Could be a single LineItem or multiple via nesting
+      if id = params[:line_items_attributes][:id].present?
+        line_items_to_destroy_ids << id if params[:line_items_attributes][:quantity].to_i <= 0
+      else
+        params[:line_items_attributes].reject! do |_index, attributes|
+          line_items_to_destroy_ids << attributes[:id] if attributes[:quantity].to_i <= 0
+        end
```

Comment:
In which cases do we have this difference in passing the attributes?

Reply:
Do you mean whether it's nested or not? Like so:
```ruby
# Case 1:
line_item_attributes: {
  id: 1,
  quantity: 1
}

# Case 2:
line_item_attributes: {
  "01" => {
    id: 1,
    quantity: 1
  },
  "02" => {
    id: 3,
    quantity: 0
  }
}
```

I am supporting single nesting for `Api::LineItemsController`, though it seems that Rails supports these different levels by default - though I couldn't find any explicit documentation about it 🤔

Reply:
Hey @RyanofWoods, I'm not able to detect any code in the `Api::LineItemsController` or its specs that seem to support this way of passing attributes. Can you please help me adding more details about your discovery?

Reply:
Case 1 is needed for `Spree::Api::LineItemsController`:
https://github.com/solidusio/solidus/blob/834347e6477ecb0b64c622d7c9c574050a41b804/api/app/controllers/spree/api/line_items_controller.rb#L56-L62

Case 2 was needed for how the OrderContents spec is setup.

Both these formats are supported by the nested_attributes in Rails.